Source/FullScreenPokemon.ts

```diff
@@ -3211,7 +3211,7 @@ module FullScreenPokemon {
             thing.numquads = 0;
             thing.movement = undefined;
 
-            if (thing.FSP) {
+            if (this) {
```

Comment:
There was one change where I changed the rest of `thing.FSP` to `this` and ran into an error where a variable was undefined (couldn't figure out why the call in that situation led to undefined variables). So while I never ran into an issue of an object not being deleted, there was at least one case where changing following `thing.FSP` to `this` would result in an error so I reverted this change back to be safe.